### ISSUE TITLE
Need help with securing routes

### ISSUE THREAD

**Issue by bvlktech** [bvlktech/Memezoid#1]
Im fairly new to backend as only a year of exp. so ill need help via either JWT or PassportJS

**Comment by lumie31** [comment]
Hello @bvlktech, we could walk through this together. What exactly do you want to achieve?

**Comment by bvlktech** [comment]
> Im fairly new to backend as only a year of exp. so ill need help via either JWT or PassportJS

this ^^

**Comment by lumie31** [comment]
I have some experience using JWT for authentication. Let me clone and run the project over here, then we could walk through what you want to achieve

**Comment by bvlktech** [comment]
ok sounds awesome!😊

**Comment by lumie31** [comment]
Hello, I looked through the code you have so far and it's a good start. Although I have some observations and it would be a good idea to fix these going forward before adding more features like securing the routes as suggested above:

- MongoDB Atlas URL is not inputted
- The format of the current schema can be modified e.g (You should have a UserSchema instead of signup, login schemas)
- The API is not modular/ easily understandable which will be a pain going forward

I want to propose a revamp of the backend code, then more features and functionality can be added. I can start off with the revamp over here if you don't mind.

So what do you think?

**Comment by bvlktech** [comment]
Yes the atlas uri ik its not there i dnt have one yet as havent created the acct yet and sure id love to check out what u have to offer as im fairy new to backend and would love to learn all that i can

**Comment by lumie31** [comment]
Alright then. Will do the re-vamp over here and I know when I'm through, you'd understand all that's going on in the backend because it will be clearer

**Comment by bvlktech** [comment]
thank you for helping☺️🙏

**Comment by JosephmBassey** [comment]
Hi  @bvlktech I have sent in a PR concerning this issues , also done some clean up on the project structure you can take a look and give feedback.  @lumie31  please you can also check it out too, I also left some todos in the PR comment that you can add .

**Comment by bvlktech** [comment]
I saw and omgosh thats ALOT of work done before we continue do you wish to? meaning do you want to see this as an actual product for users? and do you know what this is about i was just making this for fun as something to do but just asking if anyone here wants to see this as an actual project or not?

**Comment by JosephmBassey** [comment]
Yes!, I wish to continue, let me know where I can come in to help anytime.

**Comment by bvlktech** [comment]
srry didnt mean thumb down im on mobile fat thumbs

**Comment by JosephmBassey** [comment]
Oh!, alright.. it's fine

**Comment by bvlktech** [comment]
PR merged to master and left nodes to ur TO Do's @JosephmBassey

**Comment by JosephmBassey** [comment]
Alright seen, I will  create a new issue on the next todos, or you can  create a new issues on the next features.

**Comment by lumie31** [comment]
Great job so far @JosephmBassey 🙌 Will look around and see what else I can jump on

**Comment by JosephmBassey** [comment]
Thank you @lumie31   :raised_hands:

**Comment by bvlktech** [comment]
ok i still have to build out the clients view as i only got up to the profile page @lumie31 if you want u can jump in on that? also also what are your guys thoughts on this project so far? should we make it a business after done building and split the money between all three of us or like what do u guys want to do with it going forward only because i started this as something fun to make but am slowly moving away from coding im soon ganna make a blog post on the reasons as to why via https://quipbits.com but yea

**Comment by lumie31** [comment]
@bvlktech This doesn't seem like a bad idea, but I believe what would actually drive it being a business going forward is if we identify a need for the project and solve for that need and not just being money focused from the start

**Comment by bvlktech** [comment]
well to start i just made this as something fun tbh not really any "need" or "solving" of something its just a fun social media app like instagram except just meme based although im not really into it as much anymore because im putting all my focus into https://quipbits.com if you want you can have this idea project and code to have on ur own? other wise ill just trash this repo from here on out just let me know what you would like weather u want to fork it for urself or i should just trash it
